src/core/Akka.Persistence.TCK/Query/PersistenceIdsSpec.cs

```diff
@@ -76,18 +78,23 @@ public virtual void ReadJournal_AllPersistenceIds_should_find_new_events_after_d
             var source = queries.PersistenceIds();
             var probe = source.RunWith(this.SinkProbe<string>(), Materializer);
 
+            var expected = new List<string> { "h", "i", "j" };
             probe.Within(TimeSpan.FromSeconds(10), () =>
             {
-                probe.Request(1).ExpectNext();
-                return probe.ExpectNoMsg(TimeSpan.FromMilliseconds(100));
+                expected.Remove(probe.Request(1).ExpectNext()).Should().BeTrue();
+                return probe.ExpectNoMsg(TimeSpan.FromMilliseconds(500));
             });
 
             Setup("j", 1);
-            probe.Within(TimeSpan.FromSeconds(10), () =>
-            {
-                probe.Request(5).ExpectNext();
-                return probe.ExpectNext();
-            });
+            probe.Within(TimeSpan.FromSeconds(10), () => probe.Request(5).ExpectNextUnordered(expected[0], expected[1]));
+            
+            Setup("a1", 1);
+            Thread.Sleep(TimeSpan.FromSeconds(2));
```

Comment:
This is enough to trigger the bug

Reply:
I'd prefer `Task.Delay` over `Thread.Sleep` but this is fine for now.